crates/arrow_sql_gen/src/mysql.rs

```diff
@@ -0,0 +1,265 @@
+                ColumnType::MYSQL_TYPE_LONG => {
+                    handle_primitive_type!(
+                        builder,
+                        ColumnType::MYSQL_TYPE_LONG,
+                        Int32Builder,
+                        i32,
+                        row,
+                        i
+                    )
+                }
+                ColumnType::MYSQL_TYPE_LONGLONG => {
+                    handle_primitive_type!(
+                        builder,
+                        ColumnType::MYSQL_TYPE_LONGLONG,
+                        Int64Builder,
+                        i64,
+                        row,
+                        i
+                    )
+                }
+                ColumnType::MYSQL_TYPE_FLOAT => {
+                    handle_primitive_type!(
+                        builder,
+                        ColumnType::MYSQL_TYPE_FLOAT,
+                        Float32Builder,
+                        f32,
+                        row,
+                        i
+                    )
+                }
+                ColumnType::MYSQL_TYPE_DOUBLE => {
+                    handle_primitive_type!(
+                        builder,
+                        ColumnType::MYSQL_TYPE_DOUBLE,
+                        Float64Builder,
+                        f64,
+                        row,
+                        i
+                    )
+                }
+                ColumnType::MYSQL_TYPE_VARCHAR => {
+                    handle_primitive_type!(
+                        builder,
+                        ColumnType::MYSQL_TYPE_VARCHAR,
+                        StringBuilder,
+                        String,
+                        row,
+                        i
+                    )
+                }
+                ColumnType::MYSQL_TYPE_VAR_STRING => {
+                    handle_primitive_type!(
+                        builder,
+                        ColumnType::MYSQL_TYPE_VAR_STRING,
+                        StringBuilder,
+                        String,
+                        row,
+                        i
+                    )
+                }
+                _ => unimplemented!("Unsupported column type {:?}", mysql_type),
+            }
+        }
+    }
+
+    let columns = arrow_columns_builders
+        .into_iter()
+        .filter_map(|builder| builder.map(|mut b| b.finish()))
+        .collect::<Vec<ArrayRef>>();
+    let arrow_fields = arrow_fields.into_iter().flatten().collect::<Vec<Field>>();
+    let options = &RecordBatchOptions::new().with_row_count(Some(rows.len()));
+    RecordBatch::try_new_with_options(Arc::new(Schema::new(arrow_fields)), columns, options)
+        .map_err(|err| Error::FailedToBuildRecordBatch { source: err })
+}
+
+fn map_column_to_data_type(column_type: &ColumnType) -> Option<DataType> {
+    match *column_type {
+        ColumnType::MYSQL_TYPE_NULL => Some(DataType::Null),
+        ColumnType::MYSQL_TYPE_BIT => Some(DataType::Boolean),
```

Comment:
I don't think boolean is the correct conversion for "Bit"

Reply:
https://dev.mysql.com/doc/refman/8.0/en/bit-type.html

Reply:
Yes, I think the best fit option is `UInt64` and we need to convert manually
https://arrow.apache.org/docs/cpp/api/datatype.html#_CPPv4N5arrow4Type4type6UINT64E

Reply:
Handled Bit type properly
![CleanShot 2024-04-04 at 1  51 41@2x](https://github.com/spiceai/spiceai/assets/20399045/68986620-22bf-4303-902d-1394f8d6c1bc)
If needed can map it to bool-u8-u64 depending on the length